gostackparse.go

```diff
@@ -69,17 +70,30 @@ func Parse(r io.Reader) ([]*Goroutine, []error) {
 
 	statemachine:
 		switch state {
-		case stateHeader:
+		case stateHeader, stateOriginatingFrom:
```

Comment:
NIT: Why not make this a separate `case`? Is it just to avoid duplicating the code below?

```
state = stateStackFunc
if g == nil {
    abortGoroutine("invalid goroutine header")
}
```

Reply:
Yep! The ancestors function just about the same